staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -112,6 +115,40 @@ func PodResourceNameAndContainerCompletionFunc(f cmdutil.Factory) func(*cobra.Co
 	}
 }
 
+// SetImageCompletionFunc a function to complete for kubectl set image.
+func SetImageCompletionFunc(f cmdutil.Factory) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(_ *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		var comps []string
+		directive := cobra.ShellCompDirectiveNoFileComp
+		if len(args) == 0 {
+			comps, directive = setImageResourceCompletion(f, toComplete)
+		} else if len(args) >= 1 {
+			podName := convertResourceNameToPodName(f, args[0])
+			comps = CompGetContainers(f, podName, toComplete)
+			// formatedComps to reformat contianer name and append `=` at the end of container.
```

Comment:
```suggestion
			// reformat container name and append `=` at the end of container.
```
and an example of a reformatted string?

Reply:
Thanks.